### PR TITLE
 Add DeletedTeams to BadgeState

### PR DESCRIPTION
**Pull request by joshblum**
depends on https://github.com/keybase/keybase/pull/3870

cc @cjb

### CODE REVIEW

**Comment by joshblum** [comment]
@cjb i'll let you know when the server side is deployed to test

**Comment by cjb** [comment]
@joshblum cool, cjb/DESKTOP-9784-deleted-teams-banner is an untested try at integrating (if you merge this branch into that branch)

File: go/badges/badgestate.go [hunk 1]
```diff
@@ -266,6 +277,24 @@ func (b *BadgeState) UpdateWithGregor(ctx context.Context, gstate gregor.State)
...
 				b.state.NewTeamNames = append(b.state.NewTeamNames, x.TeamName)
 			}
+		case "team.delete":
+			var body []teamDeletedBody
+			if err := json.Unmarshal(item.Body().Bytes(), &body); err != nil {
+				b.log.CDebugf(ctx, "BadgeState unmarshal error for team.delete item: %v", err)
+				continue
+			}
+			for _, x := range body {
+				if x.TeamName == "" || x.OpBy.Username == "" {
```
Comment (joshblum) [inline go/badges/badgestate.go]: @cjb should we also filter if the when `x.OpBy.Username == currentUsername`? what do the designs call for?

**Comment by cjb** [comment]
@joshblum Ah, I don't think we'd want to badge anything for a team you deleted yourself.

**Pull request merged**